tests/test_bootstrap.py

```diff
@@ -35,6 +35,39 @@ def apply_mock_storage(hass_storage):
     """Apply the storage mock."""
 
 
+@pytest.fixture(autouse=True)
+async def stop_hass(loop):
+    """Make sure all hass are stopped."""
+    orig_hass = core.HomeAssistant
+
+    created = []
+
+    def mock_hass(loop=loop):
+        hass = orig_hass(loop)
+        created.append(hass)
+        return hass
+
+    with patch("homeassistant.core.HomeAssistant", mock_hass):
+        yield
+
+    for hass in created:
+        with patch.object(loop, "stop"):
+            if hass.is_running:
```

Comment:
Even if hass is not running, we've already started an executor and might have used it. I think that we should always shut it down.

Reply:
tried that, didn't work. can't shutdown twice

Reply:
ps. it does shutdown executors three lines down if we are not running :)